drivers/base/bootvid/common.c

```diff
@@ -347,14 +499,16 @@ VidBitBlt(
     PBITMAPINFOHEADER BitmapInfoHeader;
     LONG Delta;
     PUCHAR BitmapOffset;
+    ULONG PaletteCount;
 
     /* Get the Bitmap Header */
     BitmapInfoHeader = (PBITMAPINFOHEADER)Buffer;
 
     /* Initialize the palette */
+    PaletteCount = BitmapInfoHeader->biClrUsed ?
+                   BitmapInfoHeader->biClrUsed : BV_MAX_COLORS;
     InitPaletteWithTable((PULONG)(Buffer + BitmapInfoHeader->biSize),
```

Comment:
For information, bitmaps store their palette/color as [RGBQUAD](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-rgbquad)s i.e. 0x00RRGGBB.
While (hardware) palettes prefer storing colors in 0x00BBGGRR (similar to [PALETTEENTRY](https://docs.microsoft.com/en-us/previous-versions//dd162769(v=vs.85)) / [COLORREF](https://docs.microsoft.com/en-us/windows/win32/gdi/colorref)).